lib/encoder/include/AEAT6012.h

```diff
@@ -24,16 +24,18 @@ class AEAT6012 : public Encoder {
 
   AEAT6012(const Config &config);
 
-  // Trigger a blocking encoder read and retrieve the absolute position in degrees
-  // Returns whether the read was successful
-  bool getAngleDeg(float &angle) override;
+  // Trigger a blocking encoder read and store a value for angle and velocity
+  // returns whether the read was successful
+  [[nodiscard]] bool update() override;
 
-  // Trigger a blocking encoder read and retrieve the angular velocity in degrees / second
-  // Returns whether the read was successful
-  bool getAngularVelocityDegPerSec(float &speed) override;
+  // Returns the stored angle
+  float getAngleDeg() override;
```

Comment:
the reason that there is no "nodiscard" here is bc we dont care if ppl disregard the angle. but we care very much if there was an error and it is disregarded

Reply:
agreed, nodiscard should be for error reporting